### PR TITLE
Python: several fixes (2)

### PR DESCRIPTION
**Pull request by FRidh**
###### Motivation for this change

Continuation of https://github.com/NixOS/nixpkgs/pull/28884. These fixes will also need to be backported to 17.09.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] Linux
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/development/python-modules/urllib3/default.nix [hunk 1]
```diff
@@ -22,11 +23,12 @@ buildPythonPackage rec {
 
   doCheck = false;
 
-  buildInputs = [ coverage tornado mock nose psutil pysocks ];
+  buildInputs = [ pytest mock tornado ];
```
Comment (Mic92) [inline pkgs/development/python-modules/urllib3/default.nix]:
```
checkInputs = [ pytest mock ];
```

File: pkgs/development/python-modules/coveralls/default.nix [hunk 2]
```diff
@@ -24,7 +26,8 @@ buildPythonPackage rec {
   buildInputs = [
     mock
     sh
-    pytest_27
+    pytest
+    pytestrunner
     git
   ];
```
Comment (Mic92) [inline pkgs/development/python-modules/coveralls/default.nix]:
```
checkInputs = [ pytest pytestrunner mock ];
```

File: pkgs/development/python-modules/urllib3/default.nix [hunk 3]
```diff
@@ -22,11 +23,12 @@ buildPythonPackage rec {
...
+  propagatedBuildInputs = [ pyopenssl cryptography idna certifi ipaddress pysocks ];
```
Comment (Mic92) [inline pkgs/development/python-modules/urllib3/default.nix]: Archlinux seems to have banned `certifi` from their core repos as it introduces a new ssl trust store.
(But this is just an unrelated issue)

Comment (FRidh) [reply pkgs/development/python-modules/urllib3/default.nix]: That's indeed a bigger issue. Would be nice to see a RFC on how we are going to handle certificates. I think some choices were made but they're not documented.

**Comment by sifmelcara** [comment]
Based on @Mic92 's review, I checked which package is only required for checkPhase, and just force updated my two commits at sifmelcara/nixpkgs@e64128542267fae00bff60096de402947772975c

File: pkgs/development/python-modules/urllib3/default.nix (hunk 3 above)
Comment (globin) [reply pkgs/development/python-modules/urllib3/default.nix]: cc @fpletz @grahamc

**Comment by teto** [comment]
I might have forgotten to add secretstorage as a propagatedBuildInput of the python keyring package, could you add it please ?
```
nix-shell -I ~/nixpkgs2 -p python2Packages.keyring --run "python"                                                                                          
Python 2.7.13 (default, Dec 17 2016, 20:05:07) 
[GCC 6.4.0] on linux2
Type "help", "copyright", "credits" or "license" for more information.
>>> import keyring
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/nix/store/8waaccq38z6a4jaljdz57wpmzjan0lc9-python2.7-keyring-8.4.1/lib/python2.7/site-packages/keyring/__init__.py", line 6, in <module>
    from .core import (set_keyring, get_keyring, set_password, get_password,
  File "/nix/store/8waaccq38z6a4jaljdz57wpmzjan0lc9-python2.7-keyring-8.4.1/lib/python2.7/site-packages/keyring/core.py", line 185, in <module>
    init_backend()
  File "/nix/store/8waaccq38z6a4jaljdz57wpmzjan0lc9-python2.7-keyring-8.4.1/lib/python2.7/site-packages/keyring/core.py", line 69, in init_backend
    load_config()
  File "/nix/store/8waaccq38z6a4jaljdz57wpmzjan0lc9-python2.7-keyring-8.4.1/lib/python2.7/site-packages/keyring/core.py", line 139, in load_config
    return load_keyring(keyring_name)
  File "/nix/store/8waaccq38z6a4jaljdz57wpmzjan0lc9-python2.7-keyring-8.4.1/lib/python2.7/site-packages/keyring/core.py", line 109, in load_keyring
    class_.priority
  File "/nix/store/8waaccq38z6a4jaljdz57wpmzjan0lc9-python2.7-keyring-8.4.1/lib/python2.7/site-packages/keyring/util/properties.py", line 22, in __get__
    return self.fget.__get__(None, owner)()
  File "/nix/store/8waaccq38z6a4jaljdz57wpmzjan0lc9-python2.7-keyring-8.4.1/lib/python2.7/site-packages/keyring/backends/SecretService.py", line 25, in priority
    raise RuntimeError("SecretStorage required")
RuntimeError: SecretStorage required
```
while
```
nix-shell -I ~/nixpkgs2 -p python2Packages.keyring -p python2Packages.secretstorage --run "python"                                                               02:42:18 
Python 2.7.13 (default, Dec 17 2016, 20:05:07) 
[GCC 6.4.0] on linux2
Type "help", "copyright", "credits" or "license" for more information.
>>> import keyring
>>>
```
works fine.

**Pull request merged**